Automattic/jetpack
# Blocks: iterate on the design of the Promote post-publish panel

#### Changes proposed in this Pull Request:

**Do not merge, this is waiting on #27940**

This is a follow-up to 2 PRs:

- #27928 introduced a new post-publish panel.
- #27937 introduced a new analytics tool.

This PR iterates on the post-publish panel to switch from a link to a button, and ensuring the button uses the new analytics tool.

<table>
<tr>
	<td><strong>Before</strong></td>
	<td><strong>After</strong></td>
<tr>
	<td>
<img width="317" alt="Screenshot 2022-12-15 at 15 34 20" src="https://user-images.githubusercontent.com/426388/207890303-6eee5807-01bf-4807-9675-2c62da766186.png">
</td>
	<td>
<img width="286" alt="Screenshot 2022-12-22 at 13 41 47" src="https://user-images.githubusercontent.com/426388/209137180-bdde9330-6d64-4ad4-a055-f416d8d564f2.png">

</td>
</table>

#### Other information:


#### Jetpack product discussion

* 1386-gh-dotcom-forge

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Start on a WordPress.com simple or WoA site that's connected to WordPress.com.
* Switch to this branch, or use the Jetpack Beta plugin to run this branch.
* Ensure Beta blocks are enabled on your site. If you test on a WoA site, for example, add the following: `add_filter( 'jetpack_blocks_variation', function() { return 'beta'; } );`. If you work locally, you can add 
`define( 'JETPACK_BETA_BLOCKS', true );` in your `docker/wordpress/wp-config.php`. If you use a WordPress.com simple site, just ensure that you are proxied.
* Go to Posts > Add New
* open the js console, and send the following: `localStorage.setItem( 'debug', 'dops:analytics*' );`. Refresh the page for good measure.
* In the js console settings, check "Preserve log"
* Publish a post.
* Check the post-publish sidebar panel
* Does the button look good?
* Click on the button
* Come back to the editor page, and check the js console. You should see the tracking entry matching your click.

**Note**: you may notice that when clicking on the link and landing on the Advertising page, at times the page doesn't fully load and the following error appears in the console:
`proxy eror InternalServerError: 500 status code for "GET /sites/143258936/wordads/dsp/api/v1/templates/article/urn:wpcom:post:143258936:2955"`
That is something we're discussing in 1386-gh-dotcom-forge.
